src/trackingModules/CanvasTracking.js

```diff
@@ -0,0 +1,59 @@
+export const CANVAS_EVENTS = {
+  '.djs-context-pad .entry': {
+    eventName: 'canvas.contextPad',
+    dataType: 'data-action'
+  },
+  '.djs-popup .entry': {
+    eventName: 'canvas.popupMenu',
+    dataType: 'data-id'
+  },
+  '.djs-palette-entries .entry': {
+    eventName: 'canvas.palette',
+    dataType: 'data-action'
+  }
+};
+
+export default class CanvasTracking {
+  constructor(props) {
+    this._selection = props._selection;
+
+    this.track = props.track;
+  }
+
+  trackEvent(event) {
+    let payload = {};
+
+    const element = event.target;
+    const eventType = event.type;
+
+    Object.keys(CANVAS_EVENTS).forEach(eventSelector => {
+      if (matches(element, eventSelector)) {
+        const canvasEvent = CANVAS_EVENTS[eventSelector];
+        const { eventName, dataType } = canvasEvent;
+
+        payload['name'] = `${eventName}.${eventType}`;
+        payload[`${dataType}`] = element.getAttribute(dataType);
+        payload['selection'] = this._selection.get();
+
+        this.track(payload);
+      }
+    });
+  }
+
+  enable() {
+    document.body.addEventListener('click', this.trackEvent.bind(this), true);
```

Comment:
Will this track on all bpmn-js instances on a page? Why not bind to `canvas._svg` or `diagram.getContainer()`?